src/edges_estimate/plots.py

```diff
@@ -73,14 +73,15 @@ def get_models_from_mcsamples(
     return out
 
 
-def make_residual_plot_shaded(models, freqs=None, temps=None, color=None):
+def make_residual_plot_shaded(models, freqs=None, temps=None, labels=None, color=None):
     for i, (key, freq) in enumerate(freqs.items()):
         model = models[key]["model"]
         temp = temps[key]
+        label = labels[key]
```

Comment:
This will break if `labels` is not passed. What I would do is up the top of the function, say `labels = labels or {}` to make it always a dictionary. Then down here, do `label = labels.get(key, key)`, which means the default value will be the key.